include/userobject/ElasticRecoilCrossSectionUserObject.h

```diff
@@ -0,0 +1,65 @@
+/****************************************************************/
+/*            Elastic Recoil Cross Section                      */
+/****************************************************************/
+
+#ifndef ELASTICRECOILCROSSSECTIONUSEROBJECT_H
+#define ELASTICRECOILCROSSSECTIONUSEROBJECT_H
+
+// MOOSE includes
+#include "GeneralUserObject.h"
+
+// Forward Declarations
+class ElasticRecoilCrossSectionUserObject;
+
+template <>
+InputParameters validParams<ElasticRecoilCrossSectionUserObject>();
+
+class ElasticRecoilCrossSectionUserObject : public GeneralUserObject
+{
+public:
+  ElasticRecoilCrossSectionUserObject(const InputParameters & parameters);
+
+  /**
+   * Class destructor.
+   */
+  virtual ~ElasticRecoilCrossSectionUserObject();
+
+  virtual void initialize() override;
+
+  virtual void execute() override;
+
+  virtual void finalize() override;
+
+protected:
+
+  /// Function representing the reactor neutron spectrum.
+  Function & _neutron_spectrum;
+
+  /// Function representing the epithermal spectrum scattering law.
+  Function & _scattering_law;
+
+  /// Function representing the simple elastic recoil cross section calculation.
+  Function & _simple_erxs;
```

Comment:
_simple_erxs is the function that calls the variable elastic_scattering_xs. Should I rename them?